lib/codecov.js

```diff
@@ -159,10 +161,26 @@ var upload = function(args, on_success, on_failure){
 ' \\_____\\___/ \\__,_|\\___|\\___\\___/ \\_/  \n' +
 '                                ' + version);
 
+  var token = (args.options.token || process.env.codecov_token || process.env.CODECOV_TOKEN);
+
+  if (token) {
+    query.token = token;
+  }
+
   query.yaml = ['codecov.yml', '.codecov.yml'].reduce(function (result, file) {
     return result || (fs.existsSync(path.resolve(process.cwd(), file)) ? file : undefined)
   }, undefined)
 
+  // TODO: we should set other values here too based off yaml file
+  // if a yaml file was provided, then parse it and look for the token (gh #23)
+  if (query.yaml) {
+    var yaml = YAML.load(path.resolve(process.cwd(), query.yaml));
+    if (yaml.codecov && yaml.codecov.token)
+      query.token = yaml.codecov.token;
+    else
+      throw new Error('yaml file detected but `codecov` block missing');
```

Comment:
Ok. Say you have the yaml file that has the codecov block. But inside that block the token key doesn't exist. Then the `if` statement would be falsy, therefore it will hit the `else` statement.